spec/unit/target/pod_target_spec.rb

```diff
@@ -182,6 +182,54 @@ module Pod
         @pod_target.build_product_path('$BUILT_PRODUCTS_DIR').should == '$BUILT_PRODUCTS_DIR/BananaLib/libBananaLib.a'
         @pod_target.scoped.first.build_product_path('$BUILT_PRODUCTS_DIR').should == '$BUILT_PRODUCTS_DIR/BananaLib-Pods/libBananaLib-Pods.a'
       end
+
+      it 'returns the correct header search paths' do
+        @pod_target.build_headers.add_search_path('BananaLib', Platform.ios)
+        @pod_target.sandbox.public_headers.add_search_path('BananaLib', Platform.ios)
+        header_search_paths = @pod_target.recursive_target_header_search_paths
+        header_search_paths.sort.should == [
+          '${PODS_ROOT}/Headers/Private',
```

Comment:
I am still not sold that these should end up in the header search paths but in the interest of not dramatically breaking others I kept them. The big win here is that we cut the number of header search paths if we dont need them anyway.